scip-java/src/main/scala/com/sourcegraph/scip_java/buildtools/MillBuildTool.scala

```diff
@@ -0,0 +1,110 @@
+package com.sourcegraph.scip_java.buildtools
+
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+
+import scala.jdk.CollectionConverters._
+
+import com.sourcegraph.scip_java.commands.IndexCommand
+import com.sourcegraph.scip_java.BuildInfo
+
+class MillBuildTool(index: IndexCommand) extends BuildTool("mill", index) {
+
+  override def usedInCurrentDirectory(): Boolean =
+    Files.isRegularFile(index.workingDirectory.resolve("build.sc"))
+
+  override def generateScip(): Int =
+    millVersion() match {
+      case Some(version) if isSupportedMillVersion(version) =>
+        unconditionallyGenerateScip()
+      case Some(version) =>
+        failFast(
+          s"Unsupported Mill version '${version}'. " +
+            s"To fix this problem, upgrade Mill to at least ${minimalMillVersion} and try again."
+        )
+      case None =>
+        failFast(
+          s"No Mill version detected. " +
+            s"To fix this problem, run the following command and try again: " +
+            s"echo '${minimalMillVersion}' >> .mill-version"
+        )
+    }
+
+  private def failFast(message: String): Int = {
+    index.app.error(message)
+    1
+  }
+
+  private def unconditionallyGenerateScip(): Int = {
+    val localMill = Files.isRegularFile(millFile)
+    val command =
+      if (localMill) {
+        "./mill"
```

Comment:
This is also one thing I need to test first locally to ensure proc'ing like this with `./mill` actually works. I figured it's a good idea to try and just use the local mill if they have one.